### ISSUE TITLE
Xin Mo 2player controller reads a lot of buttons

### ISSUE THREAD

**Issue by ostroso** [libretro/mame#17]
Hello,
I'm using a Xin Mo 2player controller for a mame arcade machine with Retroarch.

The base system is on archlinux, with minimal X startup, emulationstation and Retroarch with Finalburn, Bsnes, MAME and others.

The problem is: when i configure the controller in retroarch menu it reads them correctly, while if i configure them inside a game in libretro-mame it reads multiple inputs, also keyboard input.

For example, the player 2 "A" button configured in Retroarch, in mame it fires also "Keyboard F2". That is the Service mode or Test in many many games with the result that those games are unplayable with 2 players.

I've kinda bypassed the issue remapping all problematic keys in mame like tab, f2 reset with other keyboard inputs, but there's still a problem with Mortal Kombat 1 and 2 (haven't tested on 3/UMK) that have the F2 set to TEST switch and i can't change it anywhere (neither the config files! :( ).
Is to notice that Nba Jam TE can remap the TEST switch so it has no issues, but shouldn't it have the same midway driver of mk/mk2?

I've read somewhere that iMAME4All has an issue on recognizing more than 16 buttons, and you need to modify somewhere and recompile the source. Could it be the same for libretro-mame.so?

Somebody has similar issues? How can it be fixed?
Thanks!

**Comment by fr500** [comment]
No, thing is for the RA core we have a lot of auto mapping being done so it
can be usable with a pad out of the box. RetroPad is mapped to keypresses
in this core too.
I've been thinking about this by exposing different controllers thru the
controller interface instead of using the automapping and even had a POC to
fix this but I haven't had the time to get it done

**Comment by ostroso** [comment]
Thanks for the reply.

So i need to edit RA source for change some automapping on keys. Probably is that RA sees the controller as unique joypad instead of 2 separate pads. Do you know in which file i can find these mappings? The controller is seen as "THT Joystick Controller".

Another thing came up in my mind: I use also libretro-fba for cps and neogeo games but there are no problem with player 2 buttons. That's strange because the F2 key should open the neogeo service (and it does, if i press the keyboard). So with fba retroarch doesn't have the same mappings of libretro-mame?

Thanks for clarifications!
